app/controllers/api/providers_controller.rb

```diff
@@ -85,6 +85,13 @@ def import_vm_resource(type, id = nil, data = {})
       end
     end
 
+    def options
+      all_options = ExtManagementSystem.leaf_subclasses.inject({}) do |ao, ems|
+        ao.merge(ems.ems_type => ems.respond_to?(:provider_settings) ? ems.provider_settings : {})
```

Comment:
It's unfortunate that not all EMSs respond to this message. Though I think this if fine for now - do you think you could follow this up by trying to push this default down into the base class?